src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/EditProjectFileCommand.cs

```diff
@@ -115,22 +147,26 @@ private void TextDocument_FileActionOccurred(object sender, TextDocumentFileActi
             }));
         }
 
-        private async Task<string> GetFileAsync(string projectFileName)
+        private async Task SetupFileAsync()
+        {
+            _tempProjectPath = GetTempFileName(Path.GetFileName(_unconfiguredProject.FullPath));
+            _lastWrittenXml = await _msbuildAccessor.GetProjectXmlAsync(_unconfiguredProject).ConfigureAwait(false);
+            _fileSystem.WriteAllText(_tempProjectPath, _lastWrittenXml);
+        }
+
+        private string GetTempFileName(string projectFileName)
         {
-            string projectXml = await _msbuildAccessor.GetProjectXmlAsync(_unconfiguredProject).ConfigureAwait(false);
             string tempDirectory = _fileSystem.GetTempFileName();
             _fileSystem.CreateDirectory(tempDirectory);
-            var tempFileName = $"{tempDirectory}\\{projectFileName}";
-            _fileSystem.WriteAllText(tempFileName, projectXml);
-            return tempFileName;
+            return $"{tempDirectory}\\{projectFileName}";
         }
 
         protected string GetCommandText(IProjectTree node)
         {
             return string.Format(VSResources.EditProjectFileCommand, node.Caption, FileExtension);
         }
 
-        protected abstract string FileExtension { get; }
+        protected string FileExtension => Path.GetExtension(_unconfiguredProject.FullPath);
 
         private bool ShouldHandle(IProjectTree node) => node.IsRoot() && _projectCapabiltiesService.Contains("OpenProjectFile");
```

Comment:
Why even make the command available in the case where OpenProjectFile isn't present?